### PR TITLE
Improvements and fixes of nep11 command

### PR DESCRIPTION
**Pull request by chenzhitong**
* [x] Add TokenIds as base64 as well
* [x] Fix error output to show inter error
* [x] Make wording say [Argument(1, Description = "TokenId of NFT (Format: HEX, BASE64)")]
* [x] Make wording say [Argument(0, Description = "NFT Contract (Symbol or Script Hash)")]
* [x] Add transfernft command to batch commands
* [x] Fix show nft
* [x] Remove regular expressions
* [x] Remove support for little-endian on TokenId
* [x] Allow contract names

### CODE REVIEW

**Comment by chenzhitong** [comment]
test:

```
PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft MyNFT node1
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 426c696e6420426f78202331
TokenId(Base64): RnJhZ21lbnQgQSAjMTAwMA==, TokenId(Hex): 467261676d656e742041202331303030
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 467261676d656e742042202331303030
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 467261676d656e742043202331303030

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft MyNFT 0x46a89f1eb2a90851a01fc8068dd9e9a2b15dcbdf
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 426c696e6420426f78202331
TokenId(Base64): RnJhZ21lbnQgQSAjMTAwMA==, TokenId(Hex): 467261676d656e742041202331303030
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 467261676d656e742042202331303030
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 467261676d656e742043202331303030

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft 0x6ff1b0a3d731b676b2331dd16b91ecb8e8d97b34 node1
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 426c696e6420426f78202331
TokenId(Base64): RnJhZ21lbnQgQSAjMTAwMA==, TokenId(Hex): 467261676d656e742041202331303030
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 467261676d656e742042202331303030
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 467261676d656e742043202331303030

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft 0x6ff1b0a3d731b676b2331dd16b91ecb8e8d97b34 0x46a89f1eb2a90851a01fc8068dd9e9a2b15dcbdf
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 426c696e6420426f78202331
TokenId(Base64): RnJhZ21lbnQgQSAjMTAwMA==, TokenId(Hex): 467261676d656e742041202331303030
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 467261676d656e742042202331303030
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 467261676d656e742043202331303030
```

**Comment by chenzhitong** [comment]
more test:

```
PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft MyNFT NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 426c696e6420426f78202331
TokenId(Base64): RnJhZ21lbnQgQSAjMTAwMA==, TokenId(Hex): 467261676d656e742041202331303030
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 467261676d656e742042202331303030
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 467261676d656e742043202331303030

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft 0x6ff1b0a3d731b676b2331dd16b91ecb8e8d97b34 NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 426c696e6420426f78202331
TokenId(Base64): RnJhZ21lbnQgQSAjMTAwMA==, TokenId(Hex): 467261676d656e742041202331303030
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 467261676d656e742042202331303030
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 467261676d656e742043202331303030

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft MyNFT NdALzgyRt3SDJTcUP7bkPtViZmVABSKzf7
No NFT yet. (Contract:0x6ff1b0a3d731b676b2331dd16b91ecb8e8d97b34, Account:0x378bd7c4362f7f71faf4debf15625090cd2a32bd)
```

**Comment by chenzhitong** [comment]
Ready for review

test:
```
PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft MyNFT node1
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 0x312320786f4220646e696c42
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 0x3030303123204220746e656d67617246
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 0x3030303123204320746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRCAjMTAwMA==, TokenId(Hex): 0x3030303123204420746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRSAjMTAwMA==, TokenId(Hex): 0x3030303123204520746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRiAjMTAwMA==, TokenId(Hex): 0x3030303123204620746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRyAjMTAwMA==, TokenId(Hex): 0x3030303123204720746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSCAjMTAwMA==, TokenId(Hex): 0x3030303123204820746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSSAjMTAwMA==, TokenId(Hex): 0x3030303123204920746e656d67617246
PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft MyNFT 0x46a89f1eb2a90851a01fc8068dd9e9a2b15dcbdf
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 0x312320786f4220646e696c42
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 0x3030303123204220746e656d67617246
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 0x3030303123204320746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRCAjMTAwMA==, TokenId(Hex): 0x3030303123204420746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRSAjMTAwMA==, TokenId(Hex): 0x3030303123204520746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRiAjMTAwMA==, TokenId(Hex): 0x3030303123204620746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRyAjMTAwMA==, TokenId(Hex): 0x3030303123204720746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSCAjMTAwMA==, TokenId(Hex): 0x3030303123204820746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSSAjMTAwMA==, TokenId(Hex): 0x3030303123204920746e656d67617246
PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft MyNFT NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 0x312320786f4220646e696c42
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 0x3030303123204220746e656d67617246
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 0x3030303123204320746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRCAjMTAwMA==, TokenId(Hex): 0x3030303123204420746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRSAjMTAwMA==, TokenId(Hex): 0x3030303123204520746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRiAjMTAwMA==, TokenId(Hex): 0x3030303123204620746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRyAjMTAwMA==, TokenId(Hex): 0x3030303123204720746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSCAjMTAwMA==, TokenId(Hex): 0x3030303123204820746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSSAjMTAwMA==, TokenId(Hex): 0x3030303123204920746e656d67617246
PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft 0x6ff1b0a3d731b676b2331dd16b91ecb8e8d97b34 node1
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 0x312320786f4220646e696c42
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 0x3030303123204220746e656d67617246
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 0x3030303123204320746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRCAjMTAwMA==, TokenId(Hex): 0x3030303123204420746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRSAjMTAwMA==, TokenId(Hex): 0x3030303123204520746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRiAjMTAwMA==, TokenId(Hex): 0x3030303123204620746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRyAjMTAwMA==, TokenId(Hex): 0x3030303123204720746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSCAjMTAwMA==, TokenId(Hex): 0x3030303123204820746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSSAjMTAwMA==, TokenId(Hex): 0x3030303123204920746e656d67617246
PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft 0x6ff1b0a3d731b676b2331dd16b91ecb8e8d97b34 0x46a89f1eb2a90851a01fc8068dd9e9a2b15dcbdf
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 0x312320786f4220646e696c42
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 0x3030303123204220746e656d67617246
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 0x3030303123204320746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRCAjMTAwMA==, TokenId(Hex): 0x3030303123204420746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRSAjMTAwMA==, TokenId(Hex): 0x3030303123204520746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRiAjMTAwMA==, TokenId(Hex): 0x3030303123204620746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRyAjMTAwMA==, TokenId(Hex): 0x3030303123204720746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSCAjMTAwMA==, TokenId(Hex): 0x3030303123204820746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSSAjMTAwMA==, TokenId(Hex): 0x3030303123204920746e656d67617246
PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft 0x6ff1b0a3d731b676b2331dd16b91ecb8e8d97b34 NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 0x312320786f4220646e696c42
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 0x3030303123204220746e656d67617246
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 0x3030303123204320746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRCAjMTAwMA==, TokenId(Hex): 0x3030303123204420746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRSAjMTAwMA==, TokenId(Hex): 0x3030303123204520746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRiAjMTAwMA==, TokenId(Hex): 0x3030303123204620746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRyAjMTAwMA==, TokenId(Hex): 0x3030303123204720746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSCAjMTAwMA==, TokenId(Hex): 0x3030303123204820746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSSAjMTAwMA==, TokenId(Hex): 0x3030303123204920746e656d67617246

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp transfernft MyNFT RnJhZ21lbnQgQiAjMTAwMA== node1 NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TransferNFT Transaction 0xe3a81ace47f59c998b324ec18e9900a23ed536bcaff2482e57d80a9364f58c82 submitted
PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp transfernft MyNFT 0x3030303123204320746e656d67617246 node1 NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TransferNFT Transaction 0x83514c723b01dfecea67b3eef6ff8e9ea76f59aed6af8399d4d84b63baa9f189 submitted
PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp transfernft 0x6ff1b0a3d731b676b2331dd16b91ecb8e8d97b34 RnJhZ21lbnQgRCAjMTAwMA== node1 NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TransferNFT Transaction 0x9c08798a954be199bcbb1ecda5cf6c866fbff4b69f20820be975e5d5b915d106 submitted
PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp transfernft 0x6ff1b0a3d731b676b2331dd16b91ecb8e8d97b34 0x3030303123204520746e656d67617246 node1 NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TransferNFT Transaction 0x14f034b5fab5fbfaef20fe7e8794af0ad2281067258f0637a565c54bd1322c81 submitted
PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0>

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft NFT-NeoVerse NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 0x312320786f4220646e696c42
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 0x3030303123204220746e656d67617246
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 0x3030303123204320746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRCAjMTAwMA==, TokenId(Hex): 0x3030303123204420746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRSAjMTAwMA==, TokenId(Hex): 0x3030303123204520746e656d67617246
```

**Comment by chenzhitong** [comment]
test:
```
PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft MyNFT node1
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 0x312320786f4220646e696c42
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 0x3030303123204220746e656d67617246
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 0x3030303123204320746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRCAjMTAwMA==, TokenId(Hex): 0x3030303123204420746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRSAjMTAwMA==, TokenId(Hex): 0x3030303123204520746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRiAjMTAwMA==, TokenId(Hex): 0x3030303123204620746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRyAjMTAwMA==, TokenId(Hex): 0x3030303123204720746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSCAjMTAwMA==, TokenId(Hex): 0x3030303123204820746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSSAjMTAwMA==, TokenId(Hex): 0x3030303123204920746e656d67617246

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft MyNFT 0x46a89f1eb2a90851a01fc8068dd9e9a2b15dcbdf
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 0x312320786f4220646e696c42
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 0x3030303123204220746e656d67617246
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 0x3030303123204320746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRCAjMTAwMA==, TokenId(Hex): 0x3030303123204420746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRSAjMTAwMA==, TokenId(Hex): 0x3030303123204520746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRiAjMTAwMA==, TokenId(Hex): 0x3030303123204620746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRyAjMTAwMA==, TokenId(Hex): 0x3030303123204720746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSCAjMTAwMA==, TokenId(Hex): 0x3030303123204820746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSSAjMTAwMA==, TokenId(Hex): 0x3030303123204920746e656d67617246

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft MyNFT NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 0x312320786f4220646e696c42
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 0x3030303123204220746e656d67617246
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 0x3030303123204320746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRCAjMTAwMA==, TokenId(Hex): 0x3030303123204420746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRSAjMTAwMA==, TokenId(Hex): 0x3030303123204520746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRiAjMTAwMA==, TokenId(Hex): 0x3030303123204620746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRyAjMTAwMA==, TokenId(Hex): 0x3030303123204720746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSCAjMTAwMA==, TokenId(Hex): 0x3030303123204820746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSSAjMTAwMA==, TokenId(Hex): 0x3030303123204920746e656d67617246

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft 0x6ff1b0a3d731b676b2331dd16b91ecb8e8d97b34 node1
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 0x312320786f4220646e696c42
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 0x3030303123204220746e656d67617246
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 0x3030303123204320746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRCAjMTAwMA==, TokenId(Hex): 0x3030303123204420746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRSAjMTAwMA==, TokenId(Hex): 0x3030303123204520746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRiAjMTAwMA==, TokenId(Hex): 0x3030303123204620746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRyAjMTAwMA==, TokenId(Hex): 0x3030303123204720746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSCAjMTAwMA==, TokenId(Hex): 0x3030303123204820746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSSAjMTAwMA==, TokenId(Hex): 0x3030303123204920746e656d67617246

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft 0x6ff1b0a3d731b676b2331dd16b91ecb8e8d97b34 0x46a89f1eb2a90851a01fc8068dd9e9a2b15dcbdf
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 0x312320786f4220646e696c42
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 0x3030303123204220746e656d67617246
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 0x3030303123204320746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRCAjMTAwMA==, TokenId(Hex): 0x3030303123204420746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRSAjMTAwMA==, TokenId(Hex): 0x3030303123204520746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRiAjMTAwMA==, TokenId(Hex): 0x3030303123204620746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRyAjMTAwMA==, TokenId(Hex): 0x3030303123204720746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSCAjMTAwMA==, TokenId(Hex): 0x3030303123204820746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSSAjMTAwMA==, TokenId(Hex): 0x3030303123204920746e656d67617246

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp show nft 0x6ff1b0a3d731b676b2331dd16b91ecb8e8d97b34 NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TokenId(Base64): QmxpbmQgQm94ICMx, TokenId(Hex): 0x312320786f4220646e696c42
TokenId(Base64): RnJhZ21lbnQgQiAjMTAwMA==, TokenId(Hex): 0x3030303123204220746e656d67617246
TokenId(Base64): RnJhZ21lbnQgQyAjMTAwMA==, TokenId(Hex): 0x3030303123204320746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRCAjMTAwMA==, TokenId(Hex): 0x3030303123204420746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRSAjMTAwMA==, TokenId(Hex): 0x3030303123204520746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRiAjMTAwMA==, TokenId(Hex): 0x3030303123204620746e656d67617246
TokenId(Base64): RnJhZ21lbnQgRyAjMTAwMA==, TokenId(Hex): 0x3030303123204720746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSCAjMTAwMA==, TokenId(Hex): 0x3030303123204820746e656d67617246
TokenId(Base64): RnJhZ21lbnQgSSAjMTAwMA==, TokenId(Hex): 0x3030303123204920746e656d67617246

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp transfernft MyNFT RnJhZ21lbnQgQiAjMTAwMA== node1 NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TransferNFT Transaction 0xe3a81ace47f59c998b324ec18e9900a23ed536bcaff2482e57d80a9364f58c82 submitted

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp transfernft MyNFT 0x3030303123204320746e656d67617246 node1 NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TransferNFT Transaction 0x83514c723b01dfecea67b3eef6ff8e9ea76f59aed6af8399d4d84b63baa9f189 submitted

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp transfernft MyNFT RnJhZ21lbnQgSSAjMTAwMA== node1 node2
TransferNFT Transaction 0x99248ace1320b08ca59a5f37342818975fad7ca014d8db22c44d02bf525409f8 submitted

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp transfernft MyNFT RnJhZ21lbnQgQiAjMTAwMA== node1 0x46a89f1eb2a90851a01fc8068dd9e9a2b15dcbdf
TransferNFT Transaction 0x0dd3b3cbcfeee2dc9ee732138d44dada527011e59631390dd006c6ac695b0e15 submitted

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp transfernft 0x6ff1b0a3d731b676b2331dd16b91ecb8e8d97b34 RnJhZ21lbnQgRCAjMTAwMA== node1 NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TransferNFT Transaction 0x9c08798a954be199bcbb1ecda5cf6c866fbff4b69f20820be975e5d5b915d106 submitted

PS D:\1Code\chen\neo-express\src\neoxp\bin\Debug\net7.0> ./neoxp transfernft 0x6ff1b0a3d731b676b2331dd16b91ecb8e8d97b34 0x3030303123204520746e656d67617246 node1 NgKHVei5eA2pkkzJJFFSVB2QJLLrwpYJk5
TransferNFT Transaction 0x14f034b5fab5fbfaef20fe7e8794af0ad2281067258f0637a565c54bd1322c81 submitted
```

**Review by Liaojinghui — APPROVED** [review]

**Comment by cschuchardt88** [comment]
Just a reminder for the future. We need to add `transfernft` to batch commands.

File: src/neoxp/Commands/ShowCommand.NFT.cs [hunk 1]
```diff
@@ -28,11 +27,11 @@ public NFT(ExpressChainManagerFactory chainManagerFactory)
                 this.chainManagerFactory = chainManagerFactory;
             }
 
-            [Argument(0, Description = "Contract to show NFT of (symbol or script hash)")]
+            [Argument(0, Description = "NFT Contract (Symbol or Script Hash)")]
             [Required]
             internal string Contract { get; init; } = string.Empty;
 
-            [Argument(1, Description = "Account to show asset balance for")]
+            [Argument(1, Description = "TokenId of NFT (Format: Script Hash, Address, Wallet name)")]
```
Comment (cschuchardt88) [inline src/neoxp/Commands/ShowCommand.NFT.cs:34]: i think this should read as `Account`?

**Comment by chenzhitong** [comment]
> Just a reminder for the future. We need to add `transfernft` to batch commands.

I have added TransferNFT to BatchCommand, please check if I have added it right

**Comment by cschuchardt88** [comment]
You have to add here as well.

https://github.com/neo-project/neo-express/blob/4c1c513e6bbf7c17b1c2d26769a28d3f609da0b7/src/neoxp/Commands/BatchCommand.cs#L110-L284

**Comment by cschuchardt88** [comment]
Also here too (as subcommand)

https://github.com/neo-project/neo-express/blob/8e089112dfb3db26f829481572e37f90407b3e3d/src/neoxp/Commands/BatchCommand.BatchFileCommands.cs#L22

**Comment by chenzhitong** [comment]
@cschuchardt88 It's added.

File: src/neoxp/Commands/TransferNFTCommand.cs [hunk 2]
```diff
@@ -76,19 +76,15 @@ internal async Task<int> OnExecuteAsync(CommandLineApplication app, IConsole con
...
             {
-                hex = hex.ToLower().Trim();
-                if (!new Regex("^(0x)?([0-9a-f]{2})+$").IsMatch(hex))
-                    throw new FormatException();
-
-                if (new Regex("^([0-9a-f]{2})+$").IsMatch(hex))
+                try
                 {
-                    return Encoding.UTF8.GetString(hex.HexToBytes());
+                    return input.StartsWith("0x") ? Encoding.UTF8.GetString(input[2..].HexToBytes().Reverse().ToArray()) : Encoding.UTF8.GetString(Convert.FromBase64String(input));
```
Comment (cschuchardt88) [inline src/neoxp/Commands/TransferNFTCommand.cs:83]: Are we sure `reverse` is appropriate?

Comment (chenzhitong) [reply src/neoxp/Commands/TransferNFTCommand.cs:83]: I'm sure, it starting with 0x is big-endian hex string, need to remove 0x and revert to byte array. Same deal [here](https://github.com/neo-project/neo-express/blob/5e72e0c70944579fd82dd37d6a601a3de0f70229/src/neoxp/Commands/ShowCommand.NFT.cs#L68C25-L68C34).

Comment (cschuchardt88) [reply src/neoxp/Commands/TransferNFTCommand.cs:83]: but that is only for `UInt160`. Normally with just `hexstring` it isnt reversed. I know people complained about this. But we can leave for now.

Comment (chenzhitong) [reply src/neoxp/Commands/TransferNFTCommand.cs:83]: Maybe a note in the documentation would be nice like [this](https://github.com/neo-project/neo-dev-portal/blob/dd4616bd3d088ce8fa365c55d37f7f70b52cd77a/docs/n3/exchange/transaction.md?plain=1#L380)

```
NOTE
In Neo, hexadecimal strings are processed in big-endian order if they are preceded by 0x, or little-endian order if they are not.
```

Comment (chenzhitong) [reply src/neoxp/Commands/TransferNFTCommand.cs:83]: Yes, `show nft` is reverse too, https://github.com/neo-project/neo-express/blob/5e72e0c70944579fd82dd37d6a601a3de0f70229/src/neoxp/Commands/ShowCommand.NFT.cs#L68C25-L68C34

**Review by cschuchardt88 — APPROVED** [review]

**Pull request merged**